pallets/attesters/src/lib.rs

```diff
@@ -0,0 +1,864 @@
+                committee_size = active_set_size;
+                false
+            } else {
+                true
+            };
+
+            let current_committee = CurrentCommittee::<T>::get();
+            PreviousCommittee::<T>::put(current_committee);
+
+            let _seed = T::RandomnessSource::random_seed();
+
+            let mut shuffled_active_set = active_set;
+            for i in (1..shuffled_active_set.len()).rev() {
+                let random_value = T::RandomnessSource::random(&i.to_be_bytes());
+                let random_index = random_value
+                    .0
+                    .as_ref()
+                    .iter()
+                    .fold(0usize, |acc, &val| (acc + val as usize) % (i + 1));
+
+                if i != random_index {
+                    shuffled_active_set.swap(i, random_index);
+                }
+            }
+
+            let new_committee = shuffled_active_set
+                .into_iter()
+                .take(committee_size)
+                .collect::<Vec<T::AccountId>>();
+
+            CurrentCommittee::<T>::put(new_committee);
+
+            full_shuffle
+        }
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_initialize(n: T::BlockNumber) -> Weight {
+            // Check if a shuffling round has passed
+            if (n % T::ShufflingFrequency::get()).is_zero() {
+                // Update the active set of attesters
+                ActiveSet::<T>::put(
+                    SortedNominatedAttesters::<T>::get()
+                        .iter()
+                        .take(32)
+                        .cloned()
+                        .map(|(account_id, _balance)| account_id)
+                        .collect::<Vec<T::AccountId>>(),
+                );
+
+                // Call shuffle_committee
+                if !Self::shuffle_committee() {
+                    log::error!("Failed to shuffle committee");
+                    return T::DbWeight::get().reads_writes(2, 2)
+                }
+
+                return T::DbWeight::get().reads_writes(5, 5)
+            }
+            0
+        }
+    }
+
+    // The genesis config type.
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        phantom: PhantomData<T>,
+    }
+
+    // The default value for the genesis config type.
+    #[cfg(feature = "std")]
+    impl<T: Config> Default for GenesisConfig<T> {
+        fn default() -> Self {
+            Self {
+                phantom: Default::default(),
+            }
+        }
+    }
+
+    // The build of genesis for the pallet.
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+        fn build(&self) {}
+    }
+
+    #[derive(Clone, Encode, Decode, Eq, PartialEq, Debug, TypeInfo)]
+    pub struct AttesterInfo {
+        pub key_ed: [u8; 32],
+        pub key_ec: [u8; 33],
+        pub key_sr: [u8; 32],
+    }
```

Comment:
I think this can be better represented as an enum which is:

- `From<KeyTypeId>`
- `AsRef` for into `&[u8]`
- `Self::verify` as a function
- #[default] over ECDSA since we don't use the others right now
- `.0` as the pkey bytes